### DISCUSSION TITLE
How to reference environment variables in nuxt.config.ts?

### DISCUSSION THREAD

**Discussion by dungonfq** [nuxt/nuxt#17742] [Questions] [answered]
Hi Nuxt team, I have a question in a case I'm facing.

1. I'm trying to import user centrics script into my Nuxt 3 application. What I did is adding meta - script option into nuxt.config
```js
export default defineNuxtConfig({
  meta: {
    script: [
      {
        type: 'application/javascript',
        id: process.env.USER_CENTRICS_ID,
        src: 'https://app.usercentrics.eu/latest/main.js',
        language: 'automatic',
        async: true,
      },
    ],
  }
})
```

2. The script was imported successfully into <head> tag. But the attribute `id` is always empty
![script-user-centric-empty-id](https://user-images.githubusercontent.com/57652048/162408510-45c0768f-930f-481f-a6af-c5112bb46518.png)

3. I also read about runtime config: https://v3.nuxtjs.org/migration/runtime-config#runtime-config

My questions:
- Can we use .env variables directly in nuxt.config.ts?
- Or the only way we can is adding them into runtime config? Cause I have no idea to use runtime config directly in nuxt.config.ts
- Is there another way to import my script into Nuxt application?

Thank you all <3

**Comment by danielroe** [comment, marked as answer]
If you want to reference runtime config, then you should add it with `useHead` into a plugin rather than defining it in your `nuxt.config`.

While it does allow referencing process.env, your nuxt.config is only read in at dev/build time; it's ignored after that.

**Reply by dungonfq** [reply]
Thank @danielroe, it adds up. I will try this solution 👏🏻
